### PR TITLE
HBASE-27240 Clean up error-prone findings in hbase-http

### PR DESCRIPTION
**Pull request by apurtell**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 40s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 34s |  master passed  |
| +1 :green_heart: |  compile  |   0m 11s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 58s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 12s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 16s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 12s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 12s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 58s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 35s |  hbase-http in the patch passed.  |
|  |   |  15m 48s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4653 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 20d3794f5e06 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 80b366817e |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/1/testReport/ |
| Max. process+thread count | 252 (vs. ulimit of 30000) |
| modules | C: hbase-http U: hbase-http |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 21s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 54s |  master passed  |
| +1 :green_heart: |  compile  |   0m 15s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 47s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 14s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 43s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 15s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 15s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 40s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 12s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 43s |  hbase-http in the patch passed.  |
|  |   |  16m 10s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4653 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 3c57b426d3e8 5.4.0-96-generic #109-Ubuntu SMP Wed Jan 12 16:49:16 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 80b366817e |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/1/testReport/ |
| Max. process+thread count | 350 (vs. ulimit of 30000) |
| modules | C: hbase-http U: hbase-http |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 58s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  1s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 55s |  master passed  |
| +1 :green_heart: |  compile  |   0m 54s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 29s |  master passed  |
| +1 :green_heart: |  spotless  |   1m 21s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 49s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 38s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 49s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 49s |  hbase-http generated 0 new + 1 unchanged - 21 fixed = 1 total (was 22)  |
| -0 :warning: |  checkstyle  |   0m 27s |  hbase-http: The patch generated 1 new + 2 unchanged - 0 fixed = 3 total (was 2)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  18m  3s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.2 3.3.1.  |
| +1 :green_heart: |  spotless  |   1m 21s |  patch has no errors when running spotless:check.  |
| -1 :x: |  spotbugs  |   1m  2s |  hbase-http generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0)  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 26s |  The patch does not generate ASF License warnings.  |
|  |   |  43m  0s |   |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:hbase-http |
|  |  Impossible cast from org.apache.hadoop.hbase.http.lib.StaticUserWebFilter to org.apache.hadoop.hbase.http.lib.StaticUserWebFilter$User in org.apache.hadoop.hbase.http.lib.StaticUserWebFilter$User.equals(Object)  At StaticUserWebFilter.java:org.apache.hadoop.hbase.http.lib.StaticUserWebFilter$User in org.apache.hadoop.hbase.http.lib.StaticUserWebFilter$User.equals(Object)  At StaticUserWebFilter.java:[line 79] |
|  |  org.apache.hadoop.hbase.http.lib.StaticUserWebFilter$User.equals(Object) checks for operand being a StaticUserWebFilter  At StaticUserWebFilter.java:a StaticUserWebFilter  At StaticUserWebFilter.java:[line 76] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4653 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux b7ae7dfda418 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 80b366817e |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| checkstyle | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/1/artifact/yetus-general-check/output/diff-checkstyle-hbase-http.txt |
| spotbugs | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/1/artifact/yetus-general-check/output/new-spotbugs-hbase-http.html |
| Max. process+thread count | 64 (vs. ulimit of 30000) |
| modules | C: hbase-http U: hbase-http |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-http/src/main/java/org/apache/hadoop/hbase/http/log/LogLevel.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ private static void printUsage() {
   }
 
   public static boolean isValidProtocol(String protocol) {
-    return ((protocol.equals(PROTOCOL_HTTP) || protocol.equals(PROTOCOL_HTTPS)));
+    return (protocol.equals(PROTOCOL_HTTP) || protocol.equals(PROTOCOL_HTTPS));
```
Comment (Apache9) [inline hbase-http/src/main/java/org/apache/hadoop/hbase/http/log/LogLevel.java:88]: The outer parentheses are still useless?

File: hbase-http/src/main/java/org/apache/hadoop/hbase/http/lib/StaticUserWebFilter.java [hunk 2]
```diff
@@ -143,8 +146,7 @@ static String getUsernameFromConf(Configuration conf) {
       // since we need to split out the username from the configured UGI.
       LOG.warn(
         DEPRECATED_UGI_KEY + " should not be used. Instead, use " + HBASE_HTTP_STATIC_USER + ".");
-      String[] parts = oldStyleUgi.split(",");
-      return parts[0];
+      return Iterators.get(Splitter.on(',').split(oldStyleUgi).iterator(), 0);
```
Comment (Apache9) [inline hbase-http/src/main/java/org/apache/hadoop/hbase/http/lib/StaticUserWebFilter.java:149]: Iterables.get so we do not need to call iterator explicitly :)

File: hbase-http/src/test/java/org/apache/hadoop/hbase/http/ssl/KeyStoreTestUtil.java [hunk 3]
```diff
@@ -369,7 +372,7 @@ private static Configuration createSSLConfig(SSLFactory.Mode mode, String keysto
    * @throws IOException if there is an I/O error saving the file
    */
   public static void saveConfig(File file, Configuration conf) throws IOException {
-    Writer writer = new FileWriter(file);
+    Writer writer = Files.newWriter(file, StandardCharsets.UTF_8);
```
Comment (Apache9) [inline hbase-http/src/test/java/org/apache/hadoop/hbase/http/ssl/KeyStoreTestUtil.java:375]: Use try with resource?

**Review by Apache9 — APPROVED** [review]
Overall LGTM. No big problems. Just some minor nits.

+1.

**Comment by apurtell** [comment]
Rebase. Address final comments and fix the findbugs warning.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  6s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 34s |  master passed  |
| +1 :green_heart: |  compile  |   0m 15s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 44s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 13s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 37s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 15s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 15s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 44s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 13s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 42s |  hbase-http in the patch passed.  |
|  |   |  16m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4653 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux b388f3131cee 5.4.0-96-generic #109-Ubuntu SMP Wed Jan 12 16:49:16 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / a3eeab8c56 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/2/testReport/ |
| Max. process+thread count | 353 (vs. ulimit of 30000) |
| modules | C: hbase-http U: hbase-http |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 38s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 27s |  master passed  |
| +1 :green_heart: |  compile  |   0m 13s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 58s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 11s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 14s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 11s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 11s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 59s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 35s |  hbase-http in the patch passed.  |
|  |   |  16m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4653 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 2b91606a1a4f 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / a3eeab8c56 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/2/testReport/ |
| Max. process+thread count | 253 (vs. ulimit of 30000) |
| modules | C: hbase-http U: hbase-http |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 55s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 38s |  master passed  |
| +1 :green_heart: |  compile  |   0m 20s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m  9s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 46s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 23s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m  7s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 17s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 17s |  hbase-http generated 0 new + 1 unchanged - 21 fixed = 1 total (was 22)  |
| -0 :warning: |  checkstyle  |   0m  7s |  hbase-http: The patch generated 1 new + 2 unchanged - 0 fixed = 3 total (was 2)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  11m 53s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.2 3.3.1.  |
| +1 :green_heart: |  spotless  |   0m 41s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 26s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  8s |  The patch does not generate ASF License warnings.  |
|  |   |  26m 38s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4653 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 633b54313d54 5.4.0-1081-aws #88~18.04.1-Ubuntu SMP Thu Jun 23 16:29:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / a3eeab8c56 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| checkstyle | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/2/artifact/yetus-general-check/output/diff-checkstyle-hbase-http.txt |
| Max. process+thread count | 67 (vs. ulimit of 30000) |
| modules | C: hbase-http U: hbase-http |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/2/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 40s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 30s |  master passed  |
| +1 :green_heart: |  compile  |   0m 12s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 59s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 10s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m  7s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 11s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 11s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 59s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m  9s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 34s |  hbase-http in the patch passed.  |
|  |   |  15m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/3/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4653 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 8505b00de797 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 486d19e99f |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/3/testReport/ |
| Max. process+thread count | 252 (vs. ulimit of 30000) |
| modules | C: hbase-http U: hbase-http |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 19s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 37s |  master passed  |
| +1 :green_heart: |  compile  |   0m 15s |  master passed  |
| +1 :green_heart: |  shadedjars  |   3m 45s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 12s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 38s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 15s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 15s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   3m 40s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 13s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 43s |  hbase-http in the patch passed.  |
|  |   |  15m 43s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4653 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux d46086309f17 5.4.0-96-generic #109-Ubuntu SMP Wed Jan 12 16:49:16 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 486d19e99f |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/3/testReport/ |
| Max. process+thread count | 366 (vs. ulimit of 30000) |
| modules | C: hbase-http U: hbase-http |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 41s |  master passed  |
| +1 :green_heart: |  compile  |   0m 25s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m  9s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 50s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 27s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 56s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 22s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 22s |  hbase-http generated 0 new + 1 unchanged - 21 fixed = 1 total (was 22)  |
| -0 :warning: |  checkstyle  |   0m  7s |  hbase-http: The patch generated 1 new + 2 unchanged - 0 fixed = 3 total (was 2)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  13m 32s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.2 3.3.1.  |
| +1 :green_heart: |  spotless  |   0m 49s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   0m 27s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  8s |  The patch does not generate ASF License warnings.  |
|  |   |  30m 53s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4653 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux d1b2e5d3036d 5.4.0-1081-aws #88~18.04.1-Ubuntu SMP Thu Jun 23 16:29:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 486d19e99f |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| checkstyle | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/3/artifact/yetus-general-check/output/diff-checkstyle-hbase-http.txt |
| Max. process+thread count | 64 (vs. ulimit of 30000) |
| modules | C: hbase-http U: hbase-http |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4653/3/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**